packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -192,29 +197,55 @@ export class HttpPlugin extends BasePlugin<Http> {
         [AttributeNames.HTTP_HOSTNAME]: host,
         [AttributeNames.HTTP_METHOD]: method,
         [AttributeNames.HTTP_PATH]: options.path || '/',
-        [AttributeNames.HTTP_USER_AGENT]: userAgent || '',
       });
 
+      if (userAgent !== undefined) {
+        span.setAttribute(AttributeNames.HTTP_USER_AGENT, userAgent);
+      }
+      // In case of req.abort() is called after receiving response,
+      // req.on('close') is called before res.on('end')
+      let reqShouldCloseSpan: boolean = true;
       request.on(
         'response',
-        (response: IncomingMessage & { aborted?: boolean }) => {
+        (
+          response: IncomingMessage & { aborted?: boolean; req: ClientRequest }
+        ) => {
+          if (response.statusCode) {
+            span.setAttributes({
+              [AttributeNames.HTTP_STATUS_CODE]: response.statusCode,
+              [AttributeNames.HTTP_STATUS_TEXT]: response.statusMessage,
+            });
+          }
+          // Check if we have the original ClientRequest
+          // Otherwise it's a package that has a custom implementation
+          // custom implementation means that listeners func is not reliable (e.g superagent, request ...) provides their own way.
+          // and we can't avoid to listen to 'end' event even no 'data'/'readable' event is attached
+          if (response.req.constructor.name === 'ClientRequest') {
+            const dataFunc = response.listeners('data') as Function[];
+            const readablefunc = response.listeners('readable') as Function[];
+            // No need to listen to response events if one of these events is not attached
+            if (dataFunc.length === 0 && readablefunc.length === 0) {
```

Comment:
Is this reliable? if user code installs it's `response` listener after the plugin this check here is executed before user code got the chance to install a `data` handler or pipe the data to somewhere.

Reply:
Can you provide an example on how It could happens. Perhaps it misses a test (see related tests) ?

Reply:
I thought about following: `http.request("http://localhost:8000").on("response", onResponse)` which is from user point of view the same as `http.request("http://localhost:8000", onResponse)` but the sequence of the installed `response` listener differs. In first case OTel will not yet see `data` listeners installed inside `onResponse()`.

Reply:
I see! Thanks. I will add this test as well.

Reply:
I removed this code since it's not reliable.